handler/src/main/java/io/netty/handler/ssl/ConscryptAlpnSslEngine.java

```diff
@@ -173,4 +194,45 @@ private static boolean isConscryptEngine(SSLEngine engine, Class<?> enginesClass
     private static Method getIsConscryptMethod(Class<?> enginesClass) throws NoSuchMethodException {
         return enginesClass.getMethod("isConscrypt", SSLEngine.class);
     }
+
+    private static final class BufferAllocatorAdapter extends BufferAllocator {
+        private final ByteBufAllocator alloc;
+
+        BufferAllocatorAdapter(ByteBufAllocator alloc) {
+            this.alloc = alloc;
+        }
+
+        @Override
+        public AllocatedBuffer allocateDirectBuffer(int capacity) {
+            return new BufferAdapter(alloc.directBuffer(capacity));
+        }
+    }
+
+    private static final class BufferAdapter extends AllocatedBuffer {
+        private final ByteBuf nettyBuffer;
+        private final ByteBuffer buffer;
+
+        BufferAdapter(ByteBuf nettyBuffer) {
+            this.nettyBuffer = nettyBuffer;
+            nettyBuffer.writerIndex(nettyBuffer.capacity());
+            this.buffer = nettyBuffer.nioBuffer();
```

Comment:
@nmittler so you basically just want to get the `ByteBuffer` of the whole `ByteBuf` ?

If so use:

```java
this.buffer = nettyBuffer.nioBuffer(0, nettyBuffer.capacity());
```
 
no need to adjust the writerIndex ;)